GalaxyZooTouchTable/GalaxyZooTouchTable/Behaviors/UIElementDragBehavior.cs

```diff
@@ -0,0 +1,124 @@
+﻿using GalaxyZooTouchTable.Lib;
+using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Interactivity;
+using System.Windows.Media;
+
+namespace GalaxyZooTouchTable.Behaviors
+{
+    public class UIElementDragBehavior : Behavior<UIElement>
+    {
+        private bool isTouchDown = false;
+        List<DependencyObject> hitResultsList = new List<DependencyObject>();
+        DragCanvas DragOverlay { get; set; }
+
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+            AssociatedObject.TouchDown += AssociatedObject_TouchDown;
+            AssociatedObject.TouchLeave += AssociatedObject_TouchLeave;
+            AssociatedObject.TouchUp += AssociatedObject_TouchUp;
+        }
+
+        public static FrameworkElement FindAncestor(Type ancestorType, Visual visual)
+        {
+            while (visual != null && !ancestorType.IsInstanceOfType(visual))
+            {
+                visual = (Visual)VisualTreeHelper.GetParent(visual);
+            }
+            return visual as FrameworkElement;
+        }
+
+        private void AssociatedObject_TouchUp(object sender, TouchEventArgs e)
+        {
+            isTouchDown = false;
+        }
+
+        private void AssociatedObject_TouchDown(object sender, TouchEventArgs e)
+        {
+            isTouchDown = true;
+        }
+
+        private void AssociatedObject_TouchLeave(object sender, TouchEventArgs e)
+        {
+            if (DragOverlay == null)
+            {
+                var visual = e.OriginalSource as Visual;
+                DragOverlay = (DragCanvas)UIElementDragBehavior.FindAncestor(typeof(DragCanvas), visual);
+            }
+
+            if (isTouchDown)
+            {
+                TouchPoint touchPosition = e.GetTouchPoint(DragOverlay);
```

Comment:
For clarity purposes, it may be useful to pull this code into a function with a descriptive title. As is, it's sort of confusing what is happening here.